File: 2021W30/B0neh3ad/pgr_12899.py
'''
    124 나라의 숫자
    https://programmers.co.kr/learn/courses/30/lessons/12899?language=python3
'''
def solution(n):
    answer = ''
    digits = ('1', '2', '4')
    while n:
        answer += digits[(n-1)%3]
        n = (n-1)//3

    return answer[::-1]

File: 2021W30/fienestar/pgr_12899.cpp
/*
    124 나라의 숫자
    https://programmers.co.kr/learn/courses/30/lessons/12899
*/
#include <string>
#include <vector>

using namespace std;

string solution(int n)
{
    static string table[4] = {
        "",
        "1",
        "2",
        "4"
    };

    if(n <= 3)
        return table[n];
    else if(n % 3 == 0)
        return solution(n/3-1) + solution(3);
    else
        return solution(n/3) + solution(n%3);
}